File: redsheet-portal/src/app/database-models/v-project-module-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vProjectModuleRecord {
    constructor(vProjectModuleRecord?: vProjectModuleRecord )  {
        let properties = [ "createdBy", "createdOn", "htmlHelp", "isClosed", "moduleOrder", "note", "ownerId", "ownerName", "ownerNote", "projectId", "projectModuleId", "projectName", "projectNote", "status", "templateModuleId", "templateModuleName",];
        for (let property of properties)
            if (vProjectModuleRecord && vProjectModuleRecord[property])
                this[property] = vProjectModuleRecord[property];
    }
 
    @range(1,2147483647)
	createdBy : number =   undefined;
 
    @required()
	createdOn : Date =   undefined;
 
    @required()
	htmlHelp : string =   undefined;
 
	isClosed : boolean = false ;
 
    @range(1,2147483647)
	moduleOrder : number =   undefined;
 
    @maxLength(1000)
	note : string =   undefined;
 
    @range(1,2147483647)
	ownerId : number =   undefined;
 
    @required()
    @maxLength(201)
	ownerName : string =   undefined;
 
	ownerNote : string =   undefined;
 
	projectId : number =   0 ;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
    @maxLength(200)
	projectName : string =   undefined;
 
    @maxLength(1000)
	projectNote : string =   undefined;
 
	status : boolean = false ;
 
    @range(1,2147483647)
	templateModuleId : number =   undefined;
 
    @required()
    @maxLength(200)
	templateModuleName : string =   undefined;


}

File: RedSheet.DbEntities/Models/vProjectModuleRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vProjectModuleRecords", Schema = "dbo")]
	[Serializable]
	public partial class vProjectModuleRecord  
	{
 
	[Range(1, int.MaxValue)]
	public int CreatedBy { get; set; } 
 
	[Required]
	public System.DateTime CreatedOn { get; set; } 
 
	[Required]
	public string HtmlHelp { get; set; } 
 
	[Required]
	public bool IsClosed { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ModuleOrder { get; set; } 
 
	[MaxLength(1000)]
	public string Note { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int OwnerId { get; set; } 
 
	[Required] 
	[MaxLength(201)]
	public string OwnerName { get; set; } 

	public string OwnerNote { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ProjectName { get; set; } 
 
	[MaxLength(1000)]
	public string ProjectNote { get; set; } 
 
	[Required]
	public bool Status { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TemplateModuleId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string TemplateModuleName { get; set; } 

	public vProjectModuleRecord()
	{
    }

    }
}

